Modules/Quotas/src/Quotas.Domain/Aggregates/Challenges/Challenge.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Linq.Expressions;
+
+namespace Backbone.Modules.Quotas.Domain.Aggregates.Challenges;
+
+public class Challenge
+{
+    public const int EXPIRY_TIME_IN_MINUTES = 10;
+
+    public required string Id { get; set; }
+    public required DateTime ExpiresAt { get; set; }
+    public required string? CreatedBy { get; set; }
+
+    public DateTime GetDateOfCreation()
+    {
+        return ExpiresAt.AddMinutes(-EXPIRY_TIME_IN_MINUTES);
+    }
+
+    public static Expression<Func<Challenge, bool>> WasCreatedInIntervalBy(DateTime from, DateTime to, string identityAddress)
+    {
+        return c => c.GetDateOfCreation() > from && c.GetDateOfCreation() < to && c.CreatedBy == identityAddress;
```

Comment:
I think that this leads to client side evaluation. You shouldn't use a method in an expression.

Reply:
It doesn't lead to client side evaluation, here's the resulting query:
```
SELECT count(*)::int
FROM "Devices"."IdentityDeletionProcesses" AS i
WHERE i.CreatedAt" >= @__from_0 
AND i."CreatedAt" <= @__to_1 
AND i."IdentityAddress" = @__createdBy_2
```

Reply:
😲😲😲
Wow, how does EF Core do this? (no answer expected 😉)